File: js/algr/binsearch.js
function binsearch(arr,x) {
  var high = arr.length - 1;
  var low = 0;
  while(low <= high) {
    var mid1 = Math.floor((high+low)/2);
    var mid = parseInt((high+low)/2, 10);
    console.log('');
    console.log('1: ' + mid1);
    console.log('2: ' + mid);
    if(x > arr[mid]) {
      low = mid + 1;
    } else if(x < arr[mid]) {
      high = mid - 1;
    } else {
      return mid;
    }
  }
  return -1;
}

exports.binsearch = binsearch;

File: c/pearl/bin_search.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int a[10] = {1, 2, 3, 4, 5, 6, 7};

int bin_search(int n) {
    int low = 0, high = 6;
    while (low < high) {
        int mid = (low + high) / 2;
        if (a[mid] == n) {
            return mid;
        } else if (a[mid] > n) {
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }
    return -1;
}


int main(int argc, const char *argv[])
{
    printf("%d\n", bin_search(65));
    return 0;
}
